### ISSUE TITLE
Should the viewer pane display the page title?

### ISSUE THREAD

**Issue by seeM** [posit-dev/positron#3693]
Here's what it currently looks like:

<img width="439" alt="image" src="https://github.com/posit-dev/positron/assets/559360/82dfeb1e-8062-4d69-8640-2ba00f949a0e">

Perhaps we could use a similar treatment to the help pane – although without the history navigation?

<img width="439" alt="image" src="https://github.com/posit-dev/positron/assets/559360/c0e2c087-b8d7-4598-b77d-df7660583d30">

**Comment by nstrayer** [comment]
Do you mean adding a small header that has the title of the page? Or replacing the URL?

**Comment by seeM** [comment]
I think the URL is helpful too, so maybe a small header below that. What do you think?

**Comment by nstrayer** [comment]
Im mildly weary of taking up more vertical space, but I guess it works fine for help. Another alternative would be the title displaying over the url bar which then on hover, hides to give the user access to the url bar functionality.

**Comment by seeM** [comment]
Yeah it's not unusual in Positron/VSCode to have two action bars taking vertical space. I think I'd start to get worried at 3 though.

**Comment by jmcphers** [comment]
In #4151 I added some plumbing to show the page title instead of the URL. Here's a page with a `<title>ggplot2 demo</title>`:

<img width="682" alt="image" src="https://github.com/user-attachments/assets/d990c3b5-e345-462f-9ff2-2af7324b6782">

However, we currently only show this for HTML files / widgets, since those don't have a meaningful URL and 95% of the file paths are randomly named temp files in the temp dir.  

We also need to cheat a little to read the title, since security policy ordinarily would prevent us from being able to read the title inside the `<iframe>`; the cheat in this case is using the Electron APIs to inject some JavaScript on the page that fires a `did-load-window` postmessage from inside the iframe to let the parent know what its title is.
